applications/RANSApplication/custom_elements/convection_diffusion_reaction_cross_wind_stabilized_element.h

```diff
@@ -0,0 +1,989 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_CONVECTION_DIFFUSION_REACTION_CROSS_WIND_STABILIZED_ELEMENT_H_INCLUDED)
+#define KRATOS_CONVECTION_DIFFUSION_REACTION_CROSS_WIND_STABILIZED_ELEMENT_H_INCLUDED
+
+// System includes
+#include <cmath>
+
+// External includes
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/time_discretization.h"
+
+// Application includes
+#include "convection_diffusion_reaction_stabilization_utilities.h"
+#include "custom_utilities/rans_calculation_utilities.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template <unsigned int TDim,
+          unsigned int TNumNodes,
+          class TConvectionDiffusionReactionData>
+class ConvectionDiffusionReactionCrossWindStabilizedElement
```

Comment:
I guess this is a base conv-diff element on top of which you will build its steady version. Am I wrong?

If not, at some point this should go to the `ConvectionDifffusionApplication`, together with its steady counterpart. Just commenting to take this into consideration for the mid-term developments (I'm afraid of the inter-dependencies that this could raise).

Reply:
Hi, This is not only convection-diffusion, this is Convection, Diffusion, and Reaction. Reaction term is the reason i have to have a new stabilization methods since Convection-diffusion stabilization fails when the flow is reaction dominant ( like in RANS models). Do you still think we need to move it to convection diffusion application. I considered this when I was creating it, as ConvDiff application is specialized to Convection and Diffusion only, I thought of having this base class in RANS. What do you think?

Reply:
Let's keep it as it is for the moment. I was just raising the point to consider it in the future :wink: